romeo/src/state.rs

```diff
@@ -232,10 +238,23 @@ fn process_bitcoin_block(
 	bitcoin_height: u32,
 	block: Block,
 ) -> (State, Vec<Task>) {
-	state
-		.deposits
-		.extend(parse_deposits(config, bitcoin_height, &block));
-	state.withdrawals.extend(parse_withdrawals(config, &block));
+	let sbtc_wallet_address = config.sbtc_wallet_address();
+	state.deposits.extend(
+		parse_deposits(config, bitcoin_height, &block)
+			.into_iter()
+			.filter(|deposit| {
+				deposit.info.sbtc_wallet_address == sbtc_wallet_address
+			})
+			.collect::<Vec<_>>(),
+	);
+	state.withdrawals.extend(
+		parse_withdrawals(config, &block)
+			.into_iter()
+			.filter(|withdrawal| {
+				withdrawal.info.sbtc_wallet_address == sbtc_wallet_address
+			})
+			.collect::<Vec<_>>(),
+	);
```

Comment:
Filter implements [IntoIterator](https://doc.rust-lang.org/std/iter/struct.Filter.html#impl-IntoIterator-for-Filter%3CI,+P%3E). No need to collect into a Vec<_>.
```suggestion
		state.deposits.extend(
		parse_deposits(config, bitcoin_height, &block)
			.into_iter()
			.filter(|deposit| {
				deposit.info.sbtc_wallet_address == sbtc_wallet_address
			}),
	);
	state.withdrawals.extend(
		parse_withdrawals(config, &block)
			.into_iter()
			.filter(|withdrawal| {
				withdrawal.info.sbtc_wallet_address == sbtc_wallet_address
			}),
	);
```

Reply:
adopted